tests/kernel/timer/timer_api/src/main.c

```diff
@@ -28,15 +28,26 @@ struct timer_data {
  */
 #define INEXACT_MS_CONVERT ((CONFIG_SYS_CLOCK_TICKS_PER_SEC % MSEC_PER_SEC) != 0)
 
+#if CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT
+/* In case of custom k_busy_wait() implementation there might be a comparison between
+ * time sources based on different clocks. Thus the maximum slew error shall be taken
+ *  into account.
+ */
 #if CONFIG_NRF_RTC_TIMER
 /* On Nordic SOCs one or both of the tick and busy-wait clocks may
  * derive from sources that have slews that sum to +/- 13%.
  */
 #define BUSY_TICK_SLEW_PPM 130000U
 #else
-/* On other platforms assume the clocks are perfectly aligned. */
-#define BUSY_TICK_SLEW_PPM 0U
+/* On other platforms the maximum skew allowed is +/- 2%. */
+#define BUSY_TICK_SLEW_PPM 20000U
```

Comment:
Would prefer to make excuses based on specific platform clock behavior.  A busy_wait being "custom" doesn't necessarily mean it's inaccurate.    See the comment above about nRF, where the skew is an understood thing that can't be reasonably worked around (well, without changing the CPU clock to be a PLL based off the same crystal the timer driver uses vs. the internal oscillator, which AIUI has power impact that Nordic doesn't want to accept on the default config).

Reply:
> Would prefer to make excuses based on specific platform clock behavior. A busy_wait being "custom" doesn't necessarily mean it's inaccurate. See the comment above about nRF, where the skew is an understood thing that can't be reasonably worked around (well, without changing the CPU clock to be a PLL based off the same crystal the timer driver uses vs. the internal oscillator, which AIUI has power impact that Nordic doesn't want to accept on the default config).

Okay, how about that?